pilot/pkg/model/push_context.go

```diff
@@ -226,8 +226,30 @@ type PushRequest struct {
 	// Start represents the time a push was started. This represents the time of adding to the PushQueue.
 	// Note that this does not include time spent debouncing.
 	Start time.Time
+
+	// Reason represents the reason for requesting a push. This should only be a fixed set of values,
+	// to avoid unbounded cardinality in metrics. If this is not set, it may be automatically filled in later.
+	// There may be multiple reasons, only if there are multiple requests merge together
+	Reason []TriggerReason
 }
 
+type TriggerReason string
+
+var (
+	// Describes a push triggered by an Endpoint change
+	EndpointUpdate TriggerReason = "endpoint"
+	// Describes a push triggered by a config change
```

Comment:
Can we elaborate more on what config change means probably specify - config change for the types defined  in pkg/config/schemas except service entries?